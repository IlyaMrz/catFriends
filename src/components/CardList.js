import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
    const cardArray = robots.map((user, i) => {
        return <Card key={robots[i].id} id={robots[i].id} name={robots[i].name} email={robots[i].email} city={robots[i].address.city} />
    });
    return (
        <div>
            {cardArray}
        </div>

    );
}
export default CardList;