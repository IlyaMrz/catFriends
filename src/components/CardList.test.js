import { shallow, mount, render } from 'enzyme';
import React from 'react';
import CardList from './CardList';
import '../setupTest';

it("expect to render Card component",()=> {
    const mockRobots = [
        {
            id:1,
            name: 'Bob Tob',
            username: 'BobTop',
            email: 'bobtop@hotbob.fuck',
            address: {
                city: 'vermontton'
            }
        }
    ]
    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot()
})