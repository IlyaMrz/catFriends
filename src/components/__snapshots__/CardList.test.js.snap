// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div>
  <Card
    city="vermontton"
    email="bobtop@hotbob.fuck"
    id={1}
    key="1"
    name="Bob Tob"
  />
</div>
`;
